lib/src/history/code_history_controller.dart

```diff
@@ -115,6 +185,20 @@ class CodeHistoryController {
   }
 
   void _push() {
+    if (!lastSelection.isValid) {
+      // Do not create record for invalid selection
+      // as it is not considered to be standart user input.
+      return;
+    }
```

Comment:
Looks like a disguise for an erroneously losing selection when inserting a suggestion. Please try to preserve selection when inserting it.

Reply:
Turns out, the method insertSelectedWord worked in a weird way.